frontend/src/operator/webui/src/app/device.service.ts

```diff
@@ -18,12 +18,21 @@ export class DeviceService {
     private sanitizer: DomSanitizer
   ) {}
 
+  compareDeviceId(x: string, y: string): number {
```

Comment:
Instead of implementing it, how about using localCompare for natural order sort? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

Reply:
Sounds good! I replaced it with using localeCompare.